flake.nix

```diff
@@ -24,211 +25,299 @@
+            psql -h $TMP postgres postgres <<EOF
+              create role $DBUSER with createdb login password '$DBPASS';
+              alter user $DBUSER with superuser;
+              create database $DBNAME with owner = $DBUSER;
+              \\connect $DBNAME
+              ALTER SCHEMA public   OWNER TO $DBUSER;
+            EOF
+          '';
 
-        nixosTests = import ./nix/nixos/tests {
-          inherit pkgs;
-        };
+          # Clean up Postgres after running database tests
+          postCheck = ''
+            echo post-check
+            DBNAME=$(whoami)
+            NAME=db_schema.sql
+            mkdir -p $out/nix-support
+            echo "Dumping schema to db_schema.sql"
+            pg_dump -h $TMP -s $DBNAME > $out/$NAME
+            kill $(cat postgres.pid)
+            echo "Adding to build products..."
+            echo "file binary-dist $out/$NAME" > $out/nix-support/hydra-build-products
+          '';
 
-        checks = project.flake'.checks // {
-          inherit hlint fourmolu shellcheck;
-        } // lib.optionalAttrs hostPlatform.isLinux (prefixNamesWith "nixosTests/" nixosTests);
+          project = (nixpkgs.haskell-nix.cabalProject' ({ config, lib, ... }: rec {
+            src = ./.;
+            name = "cardano-db-sync";
+            compiler-nix-name = lib.mkDefault "ghc8107";
+            flake.variants =
+              lib.genAttrs
+                ["ghc962"]
+                (compiler-nix-name: { inherit compiler-nix-name; });
 
-        # This is used by `nix develop` to open a devShell
-        devShells.default = project.shell;
+            inputMap = {
+              "https://input-output-hk.github.io/cardano-haskell-packages" = inputs.CHaP;
+            };
 
-        scripts = flattenTree pkgs.scripts;
+            shell.buildInputs = with nixpkgs.pkgsBuildBuild; [
+              haskell-language-server
+              gitAndTools.git
+            ];
+            shell.withHoogle = true;
 
-        packages = project.exes // scripts // {
-          # Built by `nix build .`
-          default = project.exes.cardano-db-sync;
-          inherit (pkgs) dockerImage cardano-node cardano-cli cardano-smash-server-no-basic-auth;
-        } # Add checks to be able to build them individually:
-        // (prefixNamesWith "checks/" checks);
+            modules = [
+              ({ lib, pkgs, ... }: {
+                # Ignore version bounds
+                packages.katip.doExactConfig = true;
+                # Split data to reduce closure size
+                packages.ekg.components.library.enableSeparateDataOutput = true;
+                # Use our libsodium, secp256k1, libblst forks
+                packages.cardano-crypto-praos.components.library.pkgconfig =
+                  lib.mkForce [[ pkgs.libsodium-vrf ]];
+                packages.cardano-crypto-class.components.library.pkgconfig =
+                  with pkgs; lib.mkForce [[ libsodium-vrf secp256k1 libblst ]];
```

Comment:
I thought we had most of this resolved with the crypto-overlays these days? Is this still necessary?

Reply:
You're right, this is no longer necessary